ignite/services/plugin/cache.go

```diff
@@ -0,0 +1,140 @@
+package plugin
+
+import (
+	"encoding/gob"
+	"fmt"
+	"net"
+	"path"
+	"path/filepath"
+
+	hplugin "github.com/hashicorp/go-plugin"
+
+	chainconfig "github.com/ignite/cli/ignite/config"
+	"github.com/ignite/cli/ignite/pkg/cache"
+)
+
+const (
+	cacheFileName  = "ignite_plugin_cache.db"
+	cacheNamespace = "plugin.rpc.context"
+)
+
+var (
+	storage      *cache.Storage
+	storageCache *cache.Cache[ConfigContext]
+)
+
+func init() {
+	gob.Register(hplugin.ReattachConfig{})
+	gob.Register(net.UnixAddr{})
+}
+
+type ConfigContext struct {
+	Plugin hplugin.ReattachConfig
+	Addr   net.UnixAddr
+}
+
+func WritePluginConfig(pluginPath string, conf hplugin.ReattachConfig) error {
```

Comment:
The function names in this file should contain the word `Cache` I assume, because they are completely related to that.

Externally when I see a call to `plugin.WritePluginConfig()`, I can't guess this is cache related.

Reply:
good point, i will update these methods to be more clear